src/Elastic.Apm/Helpers/SystemInfoHelper.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.Helpers
+{
+	internal class SystemInfoHelper
+	{
+		private const string ContainerUidRegex = "^[0-9a-fA-F]{64}$";
+
+		private const string PodRegex = "(?:^/kubepods/[^/]+/pod([^/]+)$)|"
+			+ "(?:^/kubepods\\.slice/kubepods-[^/]+\\.slice/kubepods-[^/]+-pod([^/]+)\\.slice$)";
+
+		private static Api.System ParseContainerId(string line)
+		{
+			//Copied from the Java agent, and C#-ified
+
+			string kubernetesPodUid = null;
+
+			var fields = line.Split(':');
+			if (fields.Length != 3) return null;
+
+			var cGroupPath = fields[2];
+
+
+			var idPart = Path.GetFileName(cGroupPath);
+
+			if (string.IsNullOrWhiteSpace(idPart)) return null;
+
+			// Legacy, e.g.: /system.slice/docker-<CID>.scope
+			if (idPart.EndsWith(".scope"))
+				idPart = idPart.Substring(0, idPart.Length - ".scope".Length).Substring(idPart.IndexOf("-", StringComparison.Ordinal) + 1);
+
+			// Looking for kubernetes info
+			var dirPathPart = Directory.GetParent(cGroupPath); // Path.get (cGroupPath);  cGroupPath Path.GetPathRoot(cGroupPath);
+			if (dirPathPart != null)
+			{
+				var dir = dirPathPart.ToString();
+
+				var pattern = new Regex(PodRegex);
```

Comment:
In this case it's not that important because this code runs only once but in general it's better to construct `Regex` only once (so I would make it a field) because it needs to parse the pattern and build state machine.

Reply:
👍  Yeah, will move it out to a field.

There are ways to boost the perf. of regex, eg. we could also make it compiled, but since this is called only once I won't do that, because that [would make the 1. (and only) run slower](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=netframework-4.8#fields)

So as you suggested, we only make this a field.